source/language/types.rst

```diff
@@ -373,3 +375,126 @@ variables whose values may only be known at run time.
+| **angle**    | :gbg:`Yes` | :rbg:`No`  | :rbg:`No`  | :rbg:`No`   | :center:`-`| :gbg:`Yes` | :rbg:`No`  | :rbg:`No`  |
++--------------+------------+------------+------------+-------------+------------+------------+------------+------------+
+| **bit**      | :gbg:`Yes` | :gbg:`Yes` | :gbg:`Yes` | :rbg:`No`   | :gbg:`Yes` | :center:`-`| :rbg:`No`  | :rbg:`No`  |
++--------------+------------+------------+------------+-------------+------------+------------+------------+------------+
+| **duration** | :rbg:`No`  | :rbg:`No`  | :rbg:`No`  | :rbg:`No*`  | :rbg:`No`  | :rbg:`No`  | :center:`-`| :rbg:`No`  |
++--------------+------------+------------+------------+-------------+------------+------------+------------+------------+
+| **qubit**    | :rbg:`No`  | :rbg:`No`  | :rbg:`No`  | :rbg:`No`   | :rbg:`No`  | :rbg:`No`  | :rbg:`No`  | :center:`-`|
++--------------+------------+------------+------------+-------------+------------+------------+------------+------------+
+
+\*Note: ``duration`` values can be converted to ``float`` using the division operator. See :ref:`divideDuration`
+
+Casting from bool
+~~~~~~~~~~~~~~~~~
+
+``bool`` values cast from ``false`` to ``0.0`` and from ``true`` to ``1.0`` or 
+an equivalent representation. ``bool`` values can only be cast to ``bit[1]`` 
+(a single bit), so explicit index syntax must be given if the target ``bit``
+has more than 1 bit of precision.
+
+Casting from int/uint
+~~~~~~~~~~~~~~~~~~~~~
+
+``int[n]`` and ``uint[n]`` values cast to the standard types mimicking C99
+behavior. Casting to ``bool`` values follows the convention ``val != 0``.
+As noted above, if the value is too large to be represented in the
+target type the result is implementation-specific. Casting to ``bit[m]`` 
+is only allowed when ``m==n``. If the target 
+``bit`` has more or less precision, then explicit slicing syntax must be given.
+As noted, the conversion is done assuming a little-endian 2's complement 
+representation.
+
+Casting from float
+~~~~~~~~~~~~~~~~~~
+
+``float[n]`` values cast to the standard types mimicking C99 behavior (*e.g.* 
+discarding the fractional part for integer-type targets). As noted above, 
+if the value is too large to be represented in the 
+target type the result is implementation-specific. 
+Casting a ``float`` value to an ``angle[m]`` is accomplished by first 
+performing a modulo 2π operation on the float value. The resulting value 
+is then converted to the nearest ``angle[m]`` possible. In the event of a 
+tie between two possible nearest values the result is the one with an even 
+least significant bit (*i.e.* round to nearest, ties to even).
+
+Casting from angle
+~~~~~~~~~~~~~~~~~~
+
+``angle[n]`` values cast to ``bool`` using the convention ``val != 0.0``. 
+Casting to ``bit[m]`` values is only allowed when 
+``n==m``, otherwise explicit slicing synatx must be provided.
```

Comment:
```suggestion
``n==m``, otherwise explicit slicing syntax must be provided.
```

Reply:
Fixed in https://github.com/Qiskit/openqasm/pull/268/commits/b60fcff30e7f2e1442eebc8376b5e948cef021f8